api/dbadapters/bigquery.ts

```diff
@@ -416,3 +449,33 @@ function convertField(field: IBigQueryFieldMetadata): dataform.IField {
   }
   return result;
 }
+
+function addDescriptionToMetadata (
+  columnDescriptions: dataform.IColumnDescriptor[],
+  metadataArray: IBigQueryFieldMetadata[]
+): IBigQueryFieldMetadata[]  {
+  const findNextField = (fields: IBigQueryFieldMetadata[], path: string[]) => fields?.find((field) => field.name === path[0])
```

Comment:
TBH i'm having some trouble reading this code. i would expect:
1. loop through `metadataArray`
2. find the column descriptor that matches this column, if one exists
3. append its description
4. recurse for any nested fields

I *think* you're doing the inverse of that loop?